boards/st/nucleo_f439zi/doc/index.rst

```diff
@@ -0,0 +1,57 @@
+.. zephyr:board:: nucleo_f439zi
+
+Overview
+********
+
+The Nucleo F439ZI board features an ARM Cortex-M4 based STM32F439ZI MCU
+with a wide range of connectivity support and configurations. This SoC
+is basically a clone of the STM32F429ZI with a supplementary hardware
+cryptographic accelerator.
+
+More information about STM32F439ZI can be found here:
+
+- `STM32F439ZI on www.st.com`_
+- `STM32F439 reference manual`_
+- `STM32F439 datasheet`_
+
+Supported Features
+==================
+
+.. zephyr:board-supported-hw::
+
+This board provides the same features as the
+:zephyr:board:`nucleo_f429zi`. You can refer to it for further
+information.
```

Comment:
Nitpick: Please remove. Each board should provide an independent documentation, adn in current case, it will be provided by the generated table.